Objects/moduleobject.c

```diff
@@ -866,6 +866,42 @@ _Py_module_getattro(PyModuleObject *m, PyObject *name)
     return _Py_module_getattro_impl(m, name, 0);
 }
 
+static int
+module_setattro(PyModuleObject *mod, PyObject *name, PyObject *value)
+{
+    PyObject *res;
+    int ret;
+    assert(mod->md_dict != NULL);
+    if (value == NULL) {
```

Comment:
I would prefer to first have the setattr code path.

Reply:
Ok.  The former layout was for consistency with other code (e.g. slot_tp_setattro() in the typeobject.c or _PyObject_GenericSetAttrWithDict()).